Examples/Io/Root/src/RootParticleReader.cpp

```diff
@@ -113,67 +112,55 @@ ActsExamples::RootParticleReader::~RootParticleReader() {
-      entry = m_entryNumbers[entry];
-    }
-    m_inputChain->GetEntry(entry);
-    ACTS_INFO("Reading event: " << context.eventNumber
-                                << " stored as entry: " << entry);
-
-    unsigned int nParticles = m_particleId->size();
-
-    for (unsigned int i = 0; i < nParticles; i++) {
-      SimParticle p;
-
-      p.setProcess(static_cast<ActsFatras::ProcessType>((*m_process)[i]));
-      p.setPdg(static_cast<Acts::PdgParticle>((*m_particleType)[i]));
-      p.setCharge((*m_q)[i] * Acts::UnitConstants::e);
-      p.setMass((*m_m)[i] * Acts::UnitConstants::GeV);
-      p.setParticleId((*m_particleId)[i]);
-      p.setPosition4((*m_vx)[i] * Acts::UnitConstants::mm,
-                     (*m_vy)[i] * Acts::UnitConstants::mm,
-                     (*m_vz)[i] * Acts::UnitConstants::mm,
-                     (*m_vt)[i] * Acts::UnitConstants::mm);
-      // NOTE: depends on the normalization done in setDirection
-      p.setDirection((*m_px)[i], (*m_py)[i], (*m_pz)[i]);
-      p.setAbsoluteMomentum((*m_p)[i] * Acts::UnitConstants::GeV);
-
-      particleContainer.insert(particleContainer.end(), p);
-      priVtxCollection.push_back((*m_vertexPrimary)[i]);
-      secVtxCollection.push_back((*m_vertexSecondary)[i]);
-    }
-
-    // Write the collections to the EventStore
-    m_outputParticles(context, std::move(particleContainer));
-
-    if (!m_cfg.vertexPrimaryCollection.empty()) {
-      m_outputPrimaryVertices(context, std::move(priVtxCollection));
-    }
-
-    if (!m_cfg.vertexSecondaryCollection.empty()) {
-      m_outputSecondaryVertices(context, std::move(secVtxCollection));
-    }
+  if (m_inputChain == nullptr || context.eventNumber >= m_events) {
+    return ProcessCode::SUCCESS;
+  }
+
+  // lock the mutex
+  std::lock_guard<std::mutex> lock(m_read_mutex);
+  // now read
+
+  // The particle collection to be filled
+  SimParticleContainer particles;
+
+  // Read the correct entry
+  auto entry = context.eventNumber;
+  if (!m_cfg.orderedEvents && entry < m_entryNumbers.size()) {
+    entry = m_entryNumbers[entry];
   }
+  m_inputChain->GetEntry(entry);
+  ACTS_INFO("Reading event: " << context.eventNumber
+                              << " stored as entry: " << entry);
+
+  unsigned int nParticles = m_particleId->size();
+
+  for (unsigned int i = 0; i < nParticles; i++) {
+    SimParticle p;
+
+    p.setProcess(static_cast<ActsFatras::ProcessType>((*m_process)[i]));
+    p.setPdg(static_cast<Acts::PdgParticle>((*m_particleType)[i]));
+    p.setCharge((*m_q)[i] * Acts::UnitConstants::e);
+    p.setMass((*m_m)[i] * Acts::UnitConstants::GeV);
+    p.setParticleId((*m_particleId)[i]);
+    p.setPosition4((*m_vx)[i] * Acts::UnitConstants::mm,
+                   (*m_vy)[i] * Acts::UnitConstants::mm,
+                   (*m_vz)[i] * Acts::UnitConstants::mm,
+                   (*m_vt)[i] * Acts::UnitConstants::mm);
+    // NOTE: depends on the normalization done in setDirection
```

Comment:
I'm not sure what this comment mean here in this context. Could you annotate for documentation?

Reply:
that comment was there before but I will try to improve it